### PR TITLE
FLP-823 Apprenticeships Web - employer & provider changes

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPaymentStatus/ChangeOfPaymentStatusEmployerController.cs [hunk 1]
```diff
@@ -49,7 +50,7 @@ public async Task<IActionResult> FreezeProviderPaymentsPage(FreezeProviderPaymen
         {
             await _apprenticeshipService.FreezePayments(model.ApprenticeshipKey, model.ReasonForFreeze);
 
-            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", model.EmployerAccountId, model.ApprenticeshipHashedId?.ToUpper()) + "?showProviderPaymentsInactive=true");
+            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", model.EmployerAccountId, model.ApprenticeshipHashedId?.ToUpper()).AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ProviderPaymentsInactive));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPaymentStatus/ChangeOfPaymentStatusEmployerController.cs:53]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs [hunk 2]
```diff
@@ -84,7 +85,7 @@ public async Task<IActionResult> EmployerInitiatedSubmitChange(EmployerChangeOfP
     {
         await _apprenticeshipService.CreatePriceHistory(model.ApprenticeshipKey, "Employer", HttpContext.User.GetUserId(), null, null, model.ApprenticeshipTotalPrice, HttpUtility.HtmlEncode(model.ReasonForChangeOfPrice), model.EffectiveFromDate.Date.GetValueOrDefault());
 
-        var employerCommitmentsReturnUrl = $"{_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", model.EmployerAccountId, model.ApprenticeshipHashedId?.ToUpper())}?showChangeOfPriceRequestSent=true";
+        var employerCommitmentsReturnUrl = $"{_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", model.EmployerAccountId, model.ApprenticeshipHashedId?.ToUpper())}".AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ChangeOfPriceRequestSent);
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs:88]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs [hunk 3]
```diff
@@ -159,11 +160,11 @@ public async Task<IActionResult> ApproveOrRejectPriceChangePage(string employerA
         {
             var userId = HttpContext.User.GetUserId();
             await _apprenticeshipService.ApprovePendingPriceChange(apprenticeshipKey, userId);
-            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()) + "?showPriceChangeApproved=true");
+            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()).AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ChangeOfPriceApproved));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs:163]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs [hunk 4]
```diff
@@ -159,11 +160,11 @@ public async Task<IActionResult> ApproveOrRejectPriceChangePage(string employerA
...
         }
 
         await _apprenticeshipService.RejectPendingPriceChange(apprenticeshipKey, rejectReason.HtmlEncode());
-        return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()) + "?showPriceChangeRejected=true");
+        return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()).AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ChangeOfPriceRejected));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceEmployerController.cs:167]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs [hunk 5]
```diff
@@ -93,12 +94,12 @@ public async Task<IActionResult> ProviderInitiatedSubmitChange(ProviderChangeOfP
         if (priceChangeStatus == "Approved")
         {
             providerCommitmentsReturnUrl = _externalProviderUrlHelper.GenerateUrl(new UrlParameters
-                { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{model.ProviderReferenceNumber}/apprentices/{model.ApprenticeshipHashedId?.ToUpper()}?showChangeOfPriceAutoApproved=true" });
+                { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{model.ProviderReferenceNumber}/apprentices/{model.ApprenticeshipHashedId?.ToUpper()}"}).AppendProviderBannersToUrl(ApprenticeDetailsBanners.ChangeOfPriceAutoApproved);
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs:97]: separated out the variable but in this instance we do need the interpolation for the other existing variables

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs [hunk 6]
```diff
@@ -188,6 +189,6 @@ public async Task<IActionResult> CancelPriceChange(long ukprn, string apprentice
         }
 
         await _apprenticeshipService.CancelPendingPriceChange(apprenticeshipKey);
-        return Redirect(_externalProviderUrlHelper.GenerateUrl(new UrlParameters { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{ukprn}/apprentices/{apprenticeshipHashedId.ToUpper()}?showPriceChangeCancelled=true" }));
+        return Redirect(_externalProviderUrlHelper.GenerateUrl(new UrlParameters { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{ukprn}/apprentices/{apprenticeshipHashedId.ToUpper()}" }).AppendProviderBannersToUrl(ApprenticeDetailsBanners.ChangeOfPriceCancelled));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfPrice/ChangeOfPriceProviderController.cs:192]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs [hunk 7]
```diff
@@ -78,11 +79,11 @@ public async Task<IActionResult> ApproveOrRejectStartDateChange(string employerA
         {
             var userId = HttpContext.User.GetUserId();
             await _apprenticeshipService.ApprovePendingStartDateChange(apprenticeshipKey, userId);
-            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()) + "?showStartDateChangeApproved=true");
+            return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()).AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ChangeOfStartDateApproved));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs:82]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs [hunk 8]
```diff
@@ -78,11 +79,11 @@ public async Task<IActionResult> ApproveOrRejectStartDateChange(string employerA
...
 		}
 
 		await _apprenticeshipService.RejectPendingStartDateChange(apprenticeshipKey, rejectReason);
-		return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()) + "?showStartDateChangeRejected=true");
+		return Redirect(_externalEmployerUrlHelper.CommitmentsV2Link("ApprenticeDetails", employerAccountId, apprenticeshipHashedId.ToUpper()).AppendEmployerBannersToUrl(ApprenticeDetailsBanners.ChangeOfStartDateRejected));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateEmployerController.cs:86]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs [hunk 9]
```diff
@@ -132,8 +133,8 @@ public async Task<IActionResult> ProviderSubmitChangeDetails(ProviderChangeOfSta
         { 
             Controller = "", 
             SubDomain = Subdomains.Approvals, 
-            RelativeRoute = $"{model.ProviderReferenceNumber}/apprentices/{model.ApprenticeshipHashedId?.ToUpper()}?banners=ChangeOfStartDateSent" 
-        });
+            RelativeRoute = $"{model.ProviderReferenceNumber}/apprentices/{model.ApprenticeshipHashedId?.ToUpper()}"
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs:136]: done

File: src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs [hunk 10]
```diff
@@ -180,7 +181,7 @@ public async Task<IActionResult> CancelStartDateChange(long ukprn, string appren
 		}
 
 		await _apprenticeshipService.CancelPendingStartDateChange(apprenticeshipKey);
-		return Redirect(_externalProviderUrlHelper.GenerateUrl(new UrlParameters { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{ukprn}/apprentices/{apprenticeshipHashedId.ToUpper()}?banners=ChangeOfStartDateCancelled" }));
+		return Redirect(_externalProviderUrlHelper.GenerateUrl(new UrlParameters { Controller = "", SubDomain = Subdomains.Approvals, RelativeRoute = $"{ukprn}/apprentices/{apprenticeshipHashedId.ToUpper()}" }).AppendProviderBannersToUrl(ApprenticeDetailsBanners.ChangeOfStartDateCancelled));
```
[the remark this answers is not in the document]
Comment (tom-gough) [reply src/SFA.DAS.Apprenticeships.Web/Controllers/ChangeOfStartDate/ChangeOfStartDateProviderController.cs:184]: done

**Review by MikeV-TalentConsulting — APPROVED** [review]

**Pull request merged**